src/main/scala/is/hail/utils/IntervalTree.scala

```diff
@@ -47,6 +58,34 @@ case class Interval(start: Any, end: Any, includeStart: Boolean, includeEnd: Boo
     c > 0 || ((!this.includeStart || !other.includeEnd) && c == 0)
   }
 
+  def adjacent(pord: ExtendedOrdering, other: Interval): Boolean = {
+      (pord.equiv(this.start, other.end) && (this.includeStart || other.includeEnd)) ||
+      (pord.equiv(this.end, other.start) && (this.includeEnd || other.includeStart))
+  }
+
+  def union(pord: ExtendedOrdering, other: Interval): Option[Interval] = {
```

Comment:
yeah. maybe I should rename?

Reply:
Union is not a closed operation on intervals which necessitates Option result, but it's weird that the empty interval is considered an interval everywhere else (including in intersect) but not in union. You're also checking rechecking emptiness in `probablyOverlaps`.

Can you get rid of the match and start the conditionals with `if (probablyOverlaps(pord, other) || adjacent(pord, other))`?

How about `coalesce` instead of `union`?

Reply:
or `merge`